### ISSUE TITLE
Bug report: wrong number of arguments when installing facter 2.4.6

### ISSUE THREAD

**Issue by albix** [caskroom/homebrew-cask#27881]
#### Description of issue

I'm installing facter for the first time and it fails with a Ruby stacktrace as shown in the output below.

Affected cask: https://github.com/caskroom/homebrew-cask/blob/master/Casks/facter.rb

<details><summary>Output of `brew cask <command> --verbose`</summary>

```
==> Downloading https://downloads.puppetlabs.com/mac/facter-2.4.6.dmg
/usr/bin/curl --fail --remote-time --location --user-agent Homebrew/1.1.4-2-g9cd5a21b (Macintosh; Intel macOS 10.12.2) curl/7.51.0 https://downloads.puppetlabs.com/mac/facter-2.4.6.dmg -C 0 -o /Users/cal/Library/Caches/Homebrew/Cask/facter--2.4.6.dmg.incomplete
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  124k  100  124k    0     0   164k      0 --:--:-- --:--:-- --:--:--  164k
==> Verifying checksum for Cask facter
==> Running installer for facter; your password may be necessary.
==> Package installers may write to any location; options such as --appdir are ignored.
==> installer: Package name is facter-2.4.6
==> installer: Installing at base path /
==> installer:PHASE:Preparing for installation…
==> installer:PHASE:Preparing the disk…
==> installer:PHASE:Preparing facter-2.4.6…
==> installer:PHASE:Waiting for other installations to complete…
==> installer:PHASE:Configuring the installation…
==> installer:STATUS:
==> installer:%2.229250
==> installer:PHASE:Validating packages…
==> installer:%97.750000
==> installer: The install failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)
Error: Command failed to execute!

==> Failed command:
/usr/bin/sudo -E -- /usr/sbin/installer -pkg #<Pathname:/usr/local/Caskroom/facter/2.4.6/facter-2.4.6.pkg> -target / -verboseR

==> Standard Output of failed command:
installer: Package name is facter-2.4.6
installer: Installing at base path /
installer:PHASE:Preparing for installation…
installer:PHASE:Preparing the disk…
installer:PHASE:Preparing facter-2.4.6…
installer:PHASE:Waiting for other installations to complete…
installer:PHASE:Configuring the installation…
installer:STATUS:
installer:%2.229250
installer:PHASE:Validating packages…
installer:%97.750000
installer: The install failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)


==> Standard Error of failed command:


==> Exit status of failed command:
#<Process::Status: pid 30293 exit 1>
Error: wrong number of arguments (1 for 4)
Follow the instructions here:
  https://github.com/caskroom/homebrew-cask/blob/master/doc/reporting_bugs/pre_bug_report.md

If this doesn’t fix the problem, please report this bug:
  https://github.com/caskroom/homebrew-cask#reporting-bugs

/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/exceptions.rb:60:in `initialize'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/installer.rb:156:in `exception'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/installer.rb:156:in `raise'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/installer.rb:156:in `rescue in install_artifacts'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/installer.rb:135:in `install_artifacts'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/installer.rb:87:in `install'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli/install.rb:25:in `block in install_casks'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli/install.rb:19:in `each'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli/install.rb:19:in `install_casks'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli/install.rb:10:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli.rb:110:in `run_command'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli.rb:149:in `process'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/brew-cask.rb:8:in `<top (required)>'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
/usr/local/Homebrew/Library/Homebrew/brew.rb:24:in `require?'
/usr/local/Homebrew/Library/Homebrew/brew.rb:100:in `<main>'
Error: Kernel.exit
```
</details>

<details><summary>Output of `brew doctor`</summary>

```
Your system is ready to brew.
```
</details>

<details><summary>Output of `brew cask doctor`</summary>

```
==> macOS Release:
10.12.2
==> Hardware Architecture:
intel-64
==> Ruby Version:
2.0.0-p648
==> Ruby Path:
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
==> Homebrew Version:
Homebrew 1.1.4-2-g9cd5a21b
Homebrew/homebrew-core (git revision 9528; last commit 2016-12-16)
==> Homebrew Executable Path:
/usr/local/bin/brew
==> Homebrew Cellar Path:
/usr/local/Cellar
==> Homebrew Repository Path:
/usr/local/Homebrew
==> Homebrew Origin:
https://github.com/Homebrew/brew
==> Homebrew-Cask Version:
Homebrew-Cask 1.1.4-2-g9cd5a21b
caskroom/homebrew-cask (git revision ed93; last commit 2016-12-16)
==> Homebrew-Cask Install Location:
<NONE>
==> Homebrew-Cask Staging Location:
/usr/local/Caskroom
==> Homebrew-Cask Cached Downloads:
/Users/cal/Library/Caches/Homebrew/Cask
1 files, 124.1K (warning: run "brew cask cleanup")
==> Homebrew-Cask Default Tap Path:
/usr/local/Homebrew/Library/Taps/caskroom/homebrew-cask
==> Homebrew-Cask Alternate Cask Taps:
/usr/local/Homebrew/Library/Taps/caskroom/homebrew-versions
==> Homebrew-Cask Default Tap Cask Count:
3486
==> Contents of $LOAD_PATH:
/usr/local/Homebrew/Library/Homebrew/cask/lib
/usr/local/Homebrew/Library/Homebrew
/Library/Ruby/Site/2.0.0
/Library/Ruby/Site/2.0.0/x86_64-darwin16
/Library/Ruby/Site/2.0.0/universal-darwin16
/Library/Ruby/Site
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/vendor_ruby/2.0.0
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/vendor_ruby/2.0.0/x86_64-darwin16
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/vendor_ruby/2.0.0/universal-darwin16
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/vendor_ruby
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/x86_64-darwin16
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/universal-darwin16
==> Contents of $RUBYLIB Environment Variable:
<NONE>
==> Contents of $RUBYOPT Environment Variable:
<NONE>
==> Contents of $RUBYPATH Environment Variable:
<NONE>
==> Contents of $RBENV_VERSION Environment Variable:
<NONE>
==> Contents of $CHRUBY_VERSION Environment Variable:
<NONE>
==> Contents of $GEM_HOME Environment Variable:
<NONE>
==> Contents of $GEM_PATH Environment Variable:
<NONE>
==> Contents of $BUNDLE_PATH Environment Variable:
<NONE>
==> Contents of $PATH Environment Variable:
PATH="/usr/local/opt/rbenv/shims:/usr/local/opt/rbenv/bin:/Users/cal/bin:/usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/opt/rbenv/shims:/usr/local/opt/rbenv/bin:/Users/cal/bin:/usr/local/sbin:/usr/local/opt/go/libexec/bin:/Users/cal/code/golang/bin:/usr/local/opt/go/libexec/bin:/Users/cal/code/golang/bin:/usr/local/Homebrew/Library/Homebrew/cask/cmd:/usr/local/Homebrew/Library/Homebrew/shims/scm"
==> Contents of $SHELL Environment Variable:
SHELL="/usr/local/bin/zsh"
==> Contents of Locale Environment Variables:
LANG="en_US.UTF-8"
LC_ALL="en_US.UTF-8"
LC_CTYPE="en_US.UTF-8"
```
</details>

**Comment by reitermarkus** [comment]
Does installing it manually work?

**Comment by Numbermaniac** [comment]
I installed it on my computer manually, and it succeeded. Installing it via Cask then also worked. However, when I tried to uninstall via cask, I got this:

```
> cask uninstall facter --verbose
==> Running uninstall process for facter; your password may be necessary
==> Removing files from pkgutil Bill-of-Materials
==> chmod: Unable to change file mode on /usr/share/doc: Operation not permitted
Error: Command failed to execute!

==> Failed command:
/usr/bin/sudo -E -- /bin/chmod -- 777 #<Pathname:/usr/share/doc>

==> Standard Output of failed command:


==> Standard Error of failed command:
chmod: Unable to change file mode on /usr/share/doc: Operation not permitted


==> Exit status of failed command:
#<Process::Status: pid 3482 exit 1>
Error: Kernel.exit
```

Looks like I'm getting the same error trying to `cask reinstall facter`, too.

**Comment by albix** [comment]
I honestly haven't tried to install `facter` manually before opening this issue.

While doing so, the installer shows an incompatibility warning. It seems that the upstream package is not compatible with macOS Sierra 10.12.2.

<img width="454" alt="screen shot 2016-12-17 at 13 10 47" src="https://cloud.githubusercontent.com/assets/730278/21286672/994b1018-c45a-11e6-9730-b4c8c7c9211a.png">

**Comment by albix** [comment]
This happens on the command line

```
% sudo installer -pkg /Volumes/facter-2.4.6/facter-2.4.6.pkg -target /
installer: Package name is facter-2.4.6
installer: Installing at base path /
installer: The install failed (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.)```

**Comment by vitorgalvao** [comment]
Seems to me like we should remove this cask anyway. Looks like its [latest standalone release](https://downloads.puppetlabs.com/mac/) is `2.4.6.` (as in the cask).

However, [on its page](https://docs.puppet.com/facter/latest/) it says it’s at version `3.5` and “Included in Puppet Enterprise 2016.5”. It also says “Facter versions prior to 3.0 will go end of life December 31, 2016”.

Seems like this is no longer provided compiled and standalone. [The GitHub page](https://github.com/puppetlabs/facter) tells you how to compile it but provides no compiled downloads.
